rest/models/tickers.go

```diff
@@ -0,0 +1,298 @@
+	CUSIP                       string         `json:"cusip,omitempty"`
+	CIK                         string         `json:"cik,omitempty"`
+	CompositeFIGI               string         `json:"composite_figi,omitempty"`
+	ShareClassFIGI              string         `json:"share_class_figi,omitempty"`
+	LastUpdatedUTC              int64          `json:"last_updated_utc,omitempty"`
+	DelistedUTC                 int64          `json:"delisted_utc,omitempty"`
+	MarketCap                   float64        `json:"market_cap,omitempty"`
+	PhoneNumber                 string         `json:"phone_number,omitempty"`
+	Address                     CompanyAddress `json:"address,omitempty"` // todo: ptr?
+	Description                 string         `json:"description,omitempty"`
+	SICCode                     string         `json:"sic_code,omitempty"`
+	SICDescription              string         `json:"sic_description,omitempty"`
+	TickerRoot                  string         `json:"ticker_root,omitempty"`
+	TickerSuffix                string         `json:"ticker_suffix,omitempty"`
+	HomepageURL                 string         `json:"homepage_url,omitempty"`
+	TotalEmployees              int32          `json:"total_employees,omitempty"`
+	ListDate                    string         `json:"list_date,omitempty"`
+	Branding                    Branding       `json:"branding,omitempty"` // todo: ptr?
+	ShareClassSharesOutstanding int64          `json:"share_class_shares_outstanding,omitempty"`
+	WeightedSharesOutstanding   int64          `json:"weighted_shares_outstanding,omitempty"`
+}
+
+// CompanyAddress contains information on the physical address of a company.
+type CompanyAddress struct {
+	Address1   string `json:"address1,omitempty"`
+	Address2   string `json:"address2,omitempty"`
+	City       string `json:"city,omitempty"`
+	State      string `json:"state,omitempty"`
+	Country    string `json:"country,omitempty"`
+	PostalCode string `json:"postal_code,omitempty"`
+}
+
+// Branding contains information related to a company's brand.
+type Branding struct {
+	LogoURL     string `json:"logo_url,omitempty"`
+	IconURL     string `json:"icon_url,omitempty"`
+	AccentColor string `json:"accent_color,omitempty"`
+	LightColor  string `json:"light_color,omitempty"`
+	DarkColor   string `json:"dark_color,omitempty"`
+}
+
+// TickersResponse contains a list of reference tickers.
+type TickersResponse struct {
+	client.BaseResponse
+	Results []*TickerDetails `json:"results,omitempty"`
+}
+
+// ListTickersParams is the set of path and query parameters that are used to request reference tickers.
+type ListTickersParams struct {
+	QueryParams ListTickersQueryParams
+}
+
+// ListTickersQueryParams is the set of query parameters for requesting reference tickers.
+type ListTickersQueryParams struct {
+	TickerEQ  *string
+	TickerLT  *string
+	TickerLTE *string
+	TickerGT  *string
+	TickerGTE *string
+
+	// todo: which ones should be enums?
```

Comment:
i'm running into more query param problems as we add more endpoints:
- some endpoints expect `fx` instead of `forex` for market type
- asset class is very similar to market type but also slightly different
- ticker types lets you query on asset class but only `stocks` returns results
- some endpoints have after primary and i don't know what that is

Reply:
at least you know what `after primary` is now